workflow-manager/limiter/limiter.go

```diff
@@ -0,0 +1,69 @@
+package limiter
+
+import (
+	"fmt"
+	"sync/atomic"
+)
+
+// Limiter is similar to sync.Pool, however, it limtis how many shared
+// objects (in this case workers) there can be
+type Limiter struct {
+	workers    chan int32
+	inProgress int32
+}
+
+// Ticket is the structure that allows you to schedule jobs
+type Ticket struct {
+	limiter *Limiter
+	worker  int32
+	dirty   int32
+}
+
+// New creates a new instance of Limiter that limits how many functions can get executed without the other ones completing first
+func New(maxWorkers int32) *Limiter {
+	l := &Limiter{make(chan int32, maxWorkers), 0}
+
+	for i := int32(0); i < maxWorkers; i++ {
+		l.workers <- i
```

Comment:
IIUC, it doesn't matter what values we fill the channel with so long as there are `maxWorkers` of them. Do we fill them with distinct values just so we could someday log what worker a ticket is using?

Reply:
It just seemed like the simplest way to fill them up and giving each ticket a distinct value seemed like a nice design choice.

Reply:
Yep, just making sure I understand the significance of the values in the channel.